src/server.toit

```diff
@@ -170,33 +176,42 @@ class ResponseWriter:
     else:
       write_headers_ status_code --message=message --has_body=false
 
-  close_on_exception_ message/string -> none:
+  // Returns true if the connection was closed due to an error.
+  close_on_exception_ message/string -> bool:
+    logger_.info message
     if body_writer_:
       // We already sent a good response code, but then something went
       // wrong.  Hard close (RST) the connection to signal to the other end
       // that we failed.
       connection_.close
       connection_ = null
+      return true
     else:
       // We don't have a body writer, so perhaps we didn't send a response
       // yet.  Send a 500 to indicate an internal server error.
       write_headers_ STATUS_INTERNAL_SERVER_ERROR --message=message --has_body=false
-    logger_.info message
+      return false
 
-  close:
+  // Close the response.  We call this automatically after the block if the
+  // user's router did not call it.  Returns true if the connection was closed
+  // due to an error.
+  close -> bool:
```

Comment:
Should we split it so the usual close always returns none and have a lower-level one that returns a bool?